File: Server/models/UserProfile.js
const mongoose = require('mongoose');
const Counter = require('./Counter'); // <--- 1. Import bộ đếm

const userProfileSchema = new mongoose.Schema({
  profiles_id: {
    type: Number,
    unique: true
  },
  user_id: {
    type: Number,
    required: true,
    unique: true, 
    ref: 'User' 
  },
  height: {
    type: Number,
    min: 0    
  },
  weight: {
    type: Number,
    min: 0
  },
  body_shape: {
    type: String,
    trim: true  
  },
  skin_tone: {
    type: String,
    trim: true
  },
  style_favourite: {
    type: String,
    trim: true
  },
  colors_favourite: {
    type: String,
    trim: true
  }
}, {
  timestamps: true 
});

// tự động tăng profiles_id 
userProfileSchema.pre('save', async function () {
  const doc = this;

  if (!doc.isNew) {
    return;
  }

  const counter = await Counter.findByIdAndUpdate(
    'profiles_id',
    { $inc: { seq: 1 } },
    { new: true, upsert: true }
  );

  doc.profiles_id = counter.seq;
});



module.exports = mongoose.model('UserProfile', userProfileSchema, 'user_profiles');

File: Android/app/src/main/java/com/example/smartfashion/model/UserProfile.kt
package com.example.smartfashion.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class UserProfile(
    @SerializedName("profiles_id")
    val profilesId: Int? = null,

    @SerializedName("user_id")
    val userId: Int,

    @SerializedName("height")
    val height: Double?,

    @SerializedName("weight")
    val weight: Double?,

    @SerializedName("body_shape")
    val bodyShape: String?,

    @SerializedName("skin_tone")
    val skinTone: String?,

    @SerializedName("style_favourite")
    val styleFavourite: String?,

    @SerializedName("colors_favourite")
    val colorsFavourite: String?,

    @SerializedName("createdAt")
    val createdAt: Date? = null,

    @SerializedName("updatedAt")
    val updatedAt: Date? = null
)